core_impl/src/main/java/io/opencensus/stats/MutableViewData.java

```diff
@@ -206,41 +191,148 @@ public MutableAggregation apply(StdDev arg) {
   @VisibleForTesting
   static AggregationData createAggregationData(MutableAggregation aggregation) {
     return aggregation.match(
-        new Function<MutableSum, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableSum arg) {
-            return SumData.create(arg.getSum());
-          }
-        },
-        new Function<MutableCount, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableCount arg) {
-            return CountData.create(arg.getCount());
-          }
-        },
-        new Function<MutableHistogram, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableHistogram arg) {
-            return HistogramData.create(arg.getBucketCounts());
-          }
-        },
-        new Function<MutableRange, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableRange arg) {
-            return RangeData.create(arg.getMin(), arg.getMax());
-          }
-        },
-        new Function<MutableMean, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableMean arg) {
-            return MeanData.create(arg.getMean());
-          }
-        },
-        new Function<MutableStdDev, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableStdDev arg) {
-            return StdDevData.create(arg.getStdDev());
-          }
-        });
+        CreateSumData.INSTANCE,
+        CreateCountData.INSTANCE,
+        CreateHistogramData.INSTANCE,
+        CreateRangeData.INSTANCE,
+        CreateMeanData.INSTANCE,
+        CreateStdDevData.INSTANCE);
+  }
+  
+  // static inner Function classes
+  
+  private static final class CreateMutableSum implements Function<Sum, MutableAggregation> {
```

Comment:
Talked offline, it seems the argument above doesn't quite apply to our case here. PECS (Producer Extends Consumer Super) may be a more concrete argument that we should go with `? extends T` as the second argument here. (reference: https://stackoverflow.com/questions/2723397/what-is-pecs-producer-extends-consumer-super)